starfish/core/image/_filter/reduce.py

```diff
@@ -0,0 +1,123 @@
+
+    See Also
+    --------
+    starfish.types.Axes
+
+    """
+
+    def __init__(
+        self, dims: Iterable[Union[Axes, str]], func: Union[str, Callable] = 'max'
+    ) -> None:
+
+        self.dims = dims
+
+        # If the user provided a string, convert to callable
+        if isinstance(func, str):
+            if func == 'max':
+                func = np.amax
+            elif func == 'mean':
+                func = np.mean
+            elif func == 'sum':
+                func = np.sum
+            else:
+                raise ValueError('func should be max, mean, or sum')
+        elif callable(func):
+            warnings.warn('User-specific functions are not logged')
+        self.func = func
+
+    _DEFAULT_TESTING_PARAMETERS = {"dims": 'r', "func": 'max'}
+
+    def run(
+            self,
+            stack: ImageStack,
+            in_place: bool = False,
+            verbose: bool = False,
+            n_processes: Optional[int] = None,
+            *args,
+    ) -> ImageStack:
+        """Performs the dimension reduction with the specifed function
+
+        Parameters
+        ----------
+        stack : ImageStack
+            Stack to be filtered.
+        in_place : bool
+            if True, process ImageStack in-place, otherwise return a new stack
+        verbose : bool
+            if True, report on filtering progress (default = False)
+        n_processes : Optional[int]
+            Number of parallel processes to devote to calculating the filter
+
+        Returns
+        -------
+        ImageStack :
+            If in-place is False, return the results of filter as a new stack. Otherwise return the
+            original stack.
+
+        """
+        if not in_place:
+            stack = deepcopy(stack)
+
+        # Apply the reducing function
+        reduced = stack._data.reduce(self.func, dim=[dim.value for dim in self.dims])
+
+        # Add the reduced dims back and align with the original stack
+        reduced = reduced.expand_dims(tuple(dim.value for dim in self.dims))
+        reduced = reduced.transpose(*stack.xarray.dims)
+
+        # Construct the stack
+        stack = stack.from_numpy(reduced.values)
+
+        return stack
+
+    @staticmethod
+    @click.command("Reduce")
+    @click.option(
+        "--dims",
+        type=click.Choice(
+            [Axes.ROUND.value, Axes.CH.value, Axes.ZPLANE.value, Axes.X.value, Axes.Y.value]
+        ),
+        multiple=True,
+        help="The dimensions the Imagestack should max project over."
+             "For multiple dimensions add multiple --dims. Ex."
+             "--dims r --dims c")
+    @click.option(
+        "--func",
+        type=click.Choice(["max", "mean", "sum"]),
+        multiple=False,
+        help="The function to apply across dims"
+             "Valid function names: max, mean, sum."
+    )
+    @click.pass_context
+    def _cli(ctx, dims, func):
+        ctx.obj["component"]._cli_run(ctx, Reduce(dims, func))
```

Comment:
if you make the constructor only accept callable, you can make it support a wide variety of cases by doing:

`func = getattr(np, func)`

the CLI would require someone to use `--func amax`, as that is the only one that doesn't map exactly from the expected name to name in numpy space.

Reply:
oh. good point. huh.

Reply:
I think this is an interesting approach and I have a couple of questions:
- Is this really accepting only a callable? As I understand it, `func` would the string of a name of a callable.
- What if the user wants to define their own function to apply across an axis or specify something that is in a different library than numpy. Would we use a try/catch on `func = getattr(np, func)`? Or is there a way to check it before trying?
- I agree that the user will expect `max` even though they would mean `amax`. Perhaps we could map them together  behind the scenes  in the constructor?
- I can replace the `func` setting logic when a string is provided with `func = getattr(np, func)` and keep the ability of users to pass their own functions as well. What do you think?